packages/effect/src/Schema.ts

```diff
@@ -820,7 +820,17 @@ export const TemplateLiteralParser = <Params extends array_.NonEmptyReadonlyArra
     }
   }
   const from = TemplateLiteral(...encodedSchemas as any)
-  const re = AST.getTemplateLiteralCapturingRegExp(from.ast as AST.TemplateLiteral)
+
+  const re = from.ast instanceof AST.TemplateLiteral
+    ? AST.getTemplateLiteralCapturingRegExp(from.ast)
+    : from.ast instanceof AST.Literal
+    ? AST.getLiteralCapturingRegExp(from.ast)
+    : null
+
+  if (re === null) {
+    throw new Error("TemplateLiteral produced unexpected AST")
+  }
```

Comment:
You can use `AST.isLiteral` here, like:

```ts
const ast = from.ast as AST.TemplateLiteral | AST.Literal
const re = AST.isLiteral(ast) ? AST.getLiteralCapturingRegExp(ast) : AST.getTemplateLiteralCapturingRegExp(ast)
```

Reply:
I updated it to this, and found out a little later that `AST.TemplateLiteral | AST.Literal` is _also incorrect_. There is yet another code path where `TemplateLiteral()` returns an `AST.Union`:

```ts
S.TemplateLiteralParser(S.Literal("foo", "bar"))
```

Reply:
Maybe we should bite the bullet and create `AST.getCapturingRegExp(ast: AST.AST)`